base/expr.jl

```diff
@@ -184,7 +184,12 @@ Give a hint to the compiler that this function is worth inlining.
 
 Small functions typically do not need the `@inline` annotation,
 as the compiler does it automatically. By using `@inline` on bigger functions,
-an extra nudge can be given to the compiler to inline it.
+an extra nudge can be given to the compiler to inline it. `@inline` can
+only be used at function definitions.
+
+If `@inline` is used at the beginning of a `do` block of a function
+call, the `do` block will be marked for inlining.
```

Comment:
I think we can generalize this statement -- now `@inline` can be used at the top of a function body, and it will naturally covers a `do` block case.
Probably we want to similar phrases as `@nospecialize`'s docstring.

Same for `@noinline` docs.